superdesk/storage/desk_media_storage.py

```diff
@@ -75,6 +77,11 @@ def put(self, content, filename=None, content_type=None, metadata=None, resource
         :param str folder: Folder that the file will be stored in
         :return str: The ID that was generated for this object
         """
+        # try to determine mimetype on the server
+        determined_content_type = self._get_mimetype(content, filename)
+        if determined_content_type:
```

Comment:
imo might be interesting to log when `content_type != determined_content_type`, I would also maybe change a signature a bit to avoid that duplication:
```py
content_type = self._get_mimetype(content, filename, content_type)
```
and might skip libmagic when eg `contet_type` param starts with `image`

Reply:
I think we can also skip when content type starts with `video` and `audio`